ql/src/java/org/apache/hadoop/hive/ql/exec/tez/Utils.java

```diff
@@ -84,4 +71,31 @@ public static SplitLocationProvider getSplitLocationProvider(Configuration conf,
     }
     return splitLocationProvider;
   }
+
+  @VisibleForTesting
+  static SplitLocationProvider getCustomSplitLocationProvider(LlapRegistryService serviceRegistry, Logger LOG) throws
+      IOException {
+    LOG.info("Using LLAP instance " + serviceRegistry.getApplicationId());
+
+    Collection<LlapServiceInstance> serviceInstances =
+        serviceRegistry.getInstances().getAllInstancesOrdered(true);
+    Preconditions.checkArgument(!serviceInstances.isEmpty(),
+        "No running LLAP daemons! Please check LLAP service status and zookeeper configuration");
+    ArrayList<String> locations = new ArrayList<>(serviceInstances.size());
+    for (LlapServiceInstance serviceInstance : serviceInstances) {
+      if (LOG.isDebugEnabled()) {
+        LOG.debug("Adding " + serviceInstance.getWorkerIdentity() + " with hostname=" +
```

Comment:
Maybe that LOG could be moved down, where we actually add to locations (or better, where we don't add it if the capacity is zero). Otherwise the log might be misleading because I would assume the daemon to be part of locations when seeing it in the logs.

Reply:
Done